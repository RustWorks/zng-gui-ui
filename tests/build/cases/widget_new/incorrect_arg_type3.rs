use zero_ui::{
    layout::AngleUnits,
    widget::{background_gradient, Wgt},
    APP,
};

fn main() {
    let _scope = APP.minimal();
    let _ = Wgt! {
        background_gradient = 0.deg(), true
    };
}
