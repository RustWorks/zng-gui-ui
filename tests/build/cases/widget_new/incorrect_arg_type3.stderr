error[E0277]: the trait bound `bool: IntoVar<GradientStops>` is not satisfied
   --> cases/widget_new/incorrect_arg_type3.rs:7:40
    |
7   |         background_gradient = 0.deg(), true
    |                               -        ^^^^ the trait `IntoVar<GradientStops>` is not implemented for `bool`
    |                               |
    |                               required by a bound introduced by this call
    |
    = help: the following implementations were found:
              <bool as IntoVar<AutoSize>>
              <bool as IntoVar<CaptureMode>>
              <bool as IntoVar<Factor>>
              <bool as IntoVar<FontFeatureState>>
            and 8 others
note: required by a bound in `background_gradient_ArgsImpl::<T_axis, T_stops>::new`
   --> $WORKSPACE/zero-ui/src/properties/visual.rs
    |
    | pub fn background_gradient(child: impl UiNode, axis: impl IntoVar<LinearGradientAxis>, stops: impl IntoVar<GradientStops>) -> impl UiNode {
    |                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `background_gradient_ArgsImpl::<T_axis, T_stops>::new`
